client-js/app/elements/labrad-registry.ts

```diff
@@ -42,15 +43,76 @@ export class LabradRegistry extends polymer.Base {
                                               this.$.editValueInput);
   }
 
+
+  getDefaultSelectedItem(): number {
+    return this.path.length > 0 ? 1 : 0;
+  }
+
+
+  cursorMove(e) {
+    const offset = this.path.length > 0 ? 1 : 0;
+    const length = this.keys.length + this.dirs.length + offset;
+
+    switch (e.detail.combo) {
+      case 'up':
+        if (this.selectedIdx === null || this.selectedIdx === 0) {
+          this.set('selectedIdx', null);
+        } else {
+          this.set('selectedIdx', this.selectedIdx - 1);
+        }
+        break;
```

Comment:
Sorry for being unclear. Right now when you hit the arrow keys the selection changes _and_ the default event handling for arrow keys kicks in to scroll the view. This _sort of_ keeps the selection on screen, but they can get out of sync so that the view keeps scrolling because of the default event handling but the selection is offscreen. I think it would be best to disable the default event handling for arrow key events and then programmatically control the scrolling to make sure the selected item stays visible.